### DISCUSSION TITLE
[HELP] How to transform functions with specific decorator on it?

### DISCUSSION THREAD

**Discussion by LePichu** [swc-project/swc#7648] [Q&A] [answered]
Copying this over from Discord, I need some help transforming functions with a specific decorator on them.

> Hi, how do I check if a visited function has a specific decorator on it? I am using `visit_mut_function` where `n` is the `Function`, I want to check if `n` has a specific decorator on it? How do I go about doing so?

>  I came up with something; 
```rs
impl VisitMut for TransformVisitor {
    fn visit_mut_function(&mut self, n: &mut Function) {
        if n.decorators.len() > 0
            && n.decorators.iter().any(|dec| {
                if dec.expr.is_call() && dec.expr.as_call().is_some().to_string() == "RPC" {
                    true
                } else {
                    false
                }
            })
        {
            match &n.body {
                Some(x) => {
                    x.stmts.clear();
                },
                None => {},
            }
        }
    }
}
```
But I get a borrow checker error with `Some(x)` as `x.stmts` is immutable and behind a `&` reference, am I supposed to now call something like `self.visit_mut_block_stmt`?

**Comment by kdy1** [comment, marked as answer]
Try
```rust
            match &mut n.body {
                Some(x) => {
                    x.stmts.clear();
                },
                None => {},
            }
```
